xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/test/resources/xwiki20/specific/escaping/escaping2.test

```diff
@@ -7,8 +7,8 @@
 .#-----------------------------------------------------
 .expect|xwiki/2.0
 .#-----------------------------------------------------
-[[reference||parameter="v~~al~]]u~"e"]]
-[[image:reference||parameter="v~~al~]]u~"e"]]
+[[reference||parameter="v~~al~]~]u~"e"]]
```

Comment:
I would have preferred a more optimized version. But I guess you had this in mind too, and it was too complex for not much gain.

Reply:
I found a different fix that should be more "optimized" by matching only the first character of each string and the rest using a look ahead as we only escape the first character. This should correctly find all places that need to be escaped.

Reply:
Great, thanks for spending time on fixing this!